src/codegen/codegen_helper_visitor.cpp

```diff
@@ -152,6 +153,66 @@ void CodegenHelperVisitor::find_ion_variables(const ast::Program& node) {
     }
 }
 
+/**
+ * Find whether or not we need to emit CVODE-related code for NEURON
+ *  Notes: we generate CVODE-related code if and only if:
+ *  - there is exactly ONE block being SOLVEd
+ *  - the block is one of the following types:
+ *      - DERIVATIVE
+ *      - KINETIC
+ *      - PROCEDURE being solved with the `after_cvode`, `cvode_t`, or `cvode_t_v` methods
+ */
+void CodegenHelperVisitor::check_cvode_codegen(const ast::Program& node) {
+    // find the breakpoint block
+    const auto& breakpoint_nodes = collect_nodes(node, {AstNodeType::BREAKPOINT_BLOCK});
+
+    // do nothing if there are no BREAKPOINT nodes
+    if (breakpoint_nodes.empty()) {
+        return;
+    }
+
+    // all (global) kinetic/derivative nodes
+    const auto& kinetic_or_derivative_nodes =
+        collect_nodes(node, {AstNodeType::KINETIC_BLOCK, AstNodeType::DERIVATIVE_BLOCK});
+
+    // all (global) procedure nodes
+    const auto& procedure_nodes = collect_nodes(node, {AstNodeType::PROCEDURE_BLOCK});
+
+    // there can only be one BREAKPOINT block in the entire program
+    const auto& breakpoint_node = std::dynamic_pointer_cast<const ast::BreakpointBlock>(
+        breakpoint_nodes[0]);
```

Comment:
This should move upwards to the other breakpoint code. The comment can be expressed nicely as an `assert`.